### PR TITLE
OLMIS-7655: Added input with proposed values

### PR DESCRIPTION
**Pull request by dariapionk**
Added input which displays proposed values while writing

OLMIS-7655

[OLMIS-7655]: https://openlmis.atlassian.net/browse/OLMIS-7655?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/react-components/inputs/input-with-proposed-values.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+import  React, { useState, useEffect } from 'react';
+import Input from '../../react-components/inputs/input';
+
+
+const InputWithProposedValues = ({ data, onClick, sortFunction, displayValue, ...props }) => {
```
Comment (NikolaLaskowska) [inline src/react-components/inputs/input-with-proposed-values.jsx]: I'm thinking about this component name... maybe InputWithSuggestions is better? @sniedzielski what do you think? Do you have any other suggestions?

Comment (sniedzielski) [reply src/react-components/inputs/input-with-proposed-values.jsx]: 'InputWithSuggestions' is a better one in my opinion.

File: src/react-components/inputs/input-with-proposed-values.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    const onSelect = (value) => {
+        onClick(value);
+        setInputValue(value[displayValue]);
+        setTouched(false);
+    }
+
+    const resultsToShow = results.map((result) => {
+        return (
+            <button className='field-full-width' style={{background: 'transparent', border: '1px solid lightgray', textAlign: 'left', height: '50px'}} key={result.id} onClick={() => onSelect(result)}>
```
Comment (sniedzielski) [inline src/react-components/inputs/input-with-proposed-values.jsx]: Can we shorten this line by declaring this button tag in such way:
```
<button 
    className='field-full-width' 
    style={{background: 'transparent', border: '1px solid lightgray', textAlign: 'left', height: '50px'}} 
    key={result.id} 
    onClick={() => onSelect(result)}
>
```

File: src/react-components/inputs/input-with-proposed-values.jsx (hunk 1 above)
Comment (dariapionk) [reply src/react-components/inputs/input-with-proposed-values.jsx]: I agree with you, something was not right with this name for me, but I could not find a better proposition, I will change it

File: src/react-components/inputs/input-with-proposed-values.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                { result[displayValue] }
+            </button>
+        );
+    });
+
+    return (
+    <>
+    <Input {...props} onChange={onChange} value={inputValue} onFocus={() => {setTouched(true)}}>
```
Comment (sniedzielski) [inline src/react-components/inputs/input-with-proposed-values.jsx]: Same as above with 'button'. Can we shorten tjis line?

File: src/react-components/inputs/input-with-proposed-values.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    </Input>
+    <div>
+        { touched && resultsToShow }
+    </div>
+    </>
+    );
+};
+
+export default InputWithProposedValues;
```
Comment (sniedzielski) [inline src/react-components/inputs/input-with-proposed-values.jsx]: please add empty line at the end of file.

File: src/react-components/inputs/input-with-proposed-values.jsx (hunk 1 above)
Comment (dariapionk) [reply src/react-components/inputs/input-with-proposed-values.jsx]: Component name updated :)

File: src/react-components/inputs/input-with-proposed-values.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    const [results, setResults] = useState([]);
+
+    useEffect(() => {
+        setInputValue('');
+        setTouched(false);
+    }, [data]);
+
+    const onChange = (value) => {
+        setInputValue(value);
+        const dataToSet = sortFunction ? data.filter((element) => { return element.name.toLowerCase().indexOf(value.toLowerCase()) > -1; }).sort(sortFunction)
```
Comment (sniedzielski) [inline src/react-components/inputs/input-with-proposed-values.jsx]: I think this conditional expression is quite long. I wonder if we can add those two elements of this expression into the separate const functions:
'data.filter((element) => { return element.name.toLowerCase().indexOf(value.toLowerCase()) > -1; }).sort(sortFunction) ' 
and 
' data.filter((element) => { return element.name.toLowerCase().indexOf(value.toLowerCase()) > -1; });' 
? 
Thanks to it this code statement could be more readable.

**Review by sniedzielski — COMMENTED** [review]
@dariapionk I left minor comments regarding code style etc. Please refer to them.

File: src/react-components/inputs/input-with-proposed-values.jsx (hunk 1 above)
Comment (sniedzielski) [reply src/react-components/inputs/input-with-proposed-values.jsx]: thx

File: src/react-components/inputs/input-with-proposed-values.jsx (hunk 2 above)
Comment (sniedzielski) [reply src/react-components/inputs/input-with-proposed-values.jsx]: Also I think that maybe we can try to move 'background: 'transparent', border: '1px solid lightgray', textAlign: 'left', height: '50px'' into scss file.

Comment (sniedzielski) [reply src/react-components/inputs/input-with-proposed-values.jsx]: But this second one is only my proposition to improve. Not obligatory to do this now.

File: src/react-components/inputs/input-with-suggestions.jsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        setInputValue('');
+        setTouched(false);
+    }, [data]);
+
+    const filterValues = (values) => {
+        return values.filter((element) => { return element.name.toLowerCase().indexOf(value.toLowerCase()) > -1; })
+    }
+
+    const filterAndSortValues = (values, sortFunction) => {
+        return values.filter((element) => { return element.name.toLowerCase().indexOf(value.toLowerCase()) > -1; }).data.sort(sortFunction)
```
Comment (sniedzielski) [inline src/react-components/inputs/input-with-suggestions.jsx]: Can we try with: 
`return filterValues(values).data.sort(sortFunction)`
?

**Review by sniedzielski — APPROVED** [review]
@dariapionk LGTM. Let's wait for Nikola's feedback about this PR.

File: src/react-components/inputs/input-with-suggestions.jsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+import  React, { useState, useEffect } from 'react';
```
Comment (NikolaLaskowska) [inline src/react-components/inputs/input-with-suggestions.jsx]: Please remove one space between import and React

Comment (dariapionk) [reply src/react-components/inputs/input-with-suggestions.jsx]: done

**Pull request merged**